packages/stems/src/components/HarmonyButton/HarmonyButton.module.css

```diff
@@ -0,0 +1,220 @@
+/* ===Base Styles=== */
+.button {
+  --button-color: var(--primary);
+  --text-color: var(--static-white);
+  --overlay-color: transparent;
+  --overlay-opacity: 0;
+  align-items: center;
+  border: 1px solid var(--button-color);
+  border-radius: var(--unit-1);
+  box-sizing: border-box;
+  color: var(--text-color);
+  cursor: pointer;
+  display: inline-flex;
+  flex-shrink: 0;
+  justify-content: center;
+  overflow: hidden;
+  position: relative;
+  text-align: center;
+  user-select: none;
+  white-space: nowrap;
+}
+
+.button:focus {
+  outline: none !important;
+}
+
+/* Only add hover styles on devices which support it */
+@media (hover: hover) {
+  .button:not(.disabled):hover {
+    transition: all var(--hover);
+    transform: scale(1.04);
+  }
+}
+
+.button:not(.disabled):active {
+  transition: all var(--press);
+  transform: scale(0.98);
+}
+
+.button.disabled {
+  pointer-events: none;
+}
+
+/* Overlay used for hover/press styling */
+.button::before {
+  content: '';
+  position: absolute;
+  display: block;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: var(--overlay-color, transparent);
+  opacity: var(--overlay-opacity, 0);
+  pointer-events: none;
+}
+
+.icon path {
+  fill: currentColor;
+}
+
+.fullWidth {
+  width: 100%;
+}
+
+/* === Sizes === */
+
+/* Small */
+.small {
+  gap: var(--unit-1);
+  height: var(--unit-8);
+  padding: var(--unit-2) var(--unit-3);
+}
+
+.small .icon {
+  width: var(--unit-4);
+  height: var(--unit-4);
+}
+
+.small .text {
+  font-size: var(--font-s);
+  font-weight: var(--font-bold);
+  line-height: var(--unit-4);
+  text-transform: capitalize;
+}
+
+/* Default */
+.default {
+  gap: var(--unit-2);
+  height: var(--unit-12);
+  padding: var(--unit-3) var(--unit-6);
+}
+
+.default .icon {
+  width: var(--unit-5);
+  height: var(--unit-5);
+}
+
+.default .text {
+  font-size: var(--font-l);
+  font-weight: var(--font-bold);
+  line-height: calc(4.5 * var(--unit));
+  text-transform: capitalize;
+}
+
+/* Large */
+.large {
+  gap: var(--unit-2);
+  height: var(--unit-15);
+  padding: var(--unit-5) var(--unit-6);
+}
+
+.large .icon {
+  width: var(--unit-6);
+  height: var(--unit-6);
+}
+
+.large .text {
+  font-size: var(--font-xl);
+  font-weight: var(--font-bold);
+  line-height: var(--unit-5);
+  letter-spacing: 0.25px;
+  text-transform: uppercase;
+}
+
+/* === Color Variants === */
+
+/* Primary */
+.primary {
+  --text-color: var(--static-white);
+  --button-color: var(--primary);
+  background: var(--button-color);
+  box-shadow: var(--shadow-near);
+}
+
+.primary:hover {
+  --overlay-color: var(--static-white);
+  --overlay-opacity: 0.1;
+  box-shadow: var(--shadow-mid);
+}
+
+.primary:active {
+  --overlay-color: var(--static-black);
+  --overlay-opacity: 0.2;
+  box-shadow: none;
+}
+
+/* Secondary */
+.secondary {
+  --button-color: var(--neutral-light-5);
```

Comment:
really cool with the css-vars defined in the className. do you mind giving a quick tldr on why we do this again?

Reply:
Yep! There were two goals:
* Allowing override of the button color from JS by injecting a different value for the variable. (This is only used in primary buttons at the moment)
* Explicitly linking various properties logically to the button color. The idea is that you think about the button having a color, and then the style rules for each variant decide how the color is _applied_ to the button (ex. Secondary buttons apply the color to only the border, primary apply the color to the background).
* Defining how the color maps to the properties once, then just changing the color for hover/active states.

But I could see how this is arguably not any easier to follow than just setting the individual values per variant, per state. Open to simplifying it if you think it reads better.

Reply:
thank you for all the context, this makes sense, esp with allowing it styled through js. lets continue with this pattern and see how folks get along with it 🔥